tests/loaders/test_youtube_video.py

```diff
@@ -1,53 +1,64 @@
-        MagicMock(
-            page_content=mock_page_content,
-            metadata={"url": video_url, "title": "Test Video"},
-        )
-    ]
-
+    mock_result = {
+        'title': 'Test Video',
+        'uploader': 'Uploader Name',
+        'upload_date': '20200101',
+        'duration': 3600,
+        'view_count': 1000,
+        'like_count': 100,
+        'dislike_count': 10,
+        'average_rating': 4.5,
+        'categories': ['Education'],
+        'tags': ['tag1', 'tag2'],
+        'id': 'VIDEO_ID'
+    }
     mock_transcript = [{"text": "sample text", "start": 0.0, "duration": 5.0}]
+    serialized_transcript = json.dumps(mock_transcript, ensure_ascii=False)  # Serialize the transcript for the test
+    content = ' '.join([item['text'] for item in mock_transcript])
 
-    with patch("embedchain.loaders.youtube_video.YoutubeLoader.from_youtube_url", return_value=mock_loader), patch(
-        "embedchain.loaders.youtube_video.YouTubeTranscriptApi.get_transcript", return_value=mock_transcript
-    ):
+    with patch.object(yt_dlp.YoutubeDL, 'extract_info', return_value=mock_result), \
+         patch.object(YouTubeTranscriptApi, 'get_transcript', return_value=mock_transcript):
         result = youtube_video_loader.load_data(video_url)
 
-    expected_doc_id = hashlib.sha256((mock_page_content + video_url).encode()).hexdigest()
-
-    assert result["doc_id"] == expected_doc_id
-
-    expected_data = [
-        {
-            "content": "This is a YouTube video content.",
-            "meta_data": {"url": video_url, "title": "Test Video", "transcript": "Unavailable"},
+    expected_doc_id = hashlib.sha256((content + video_url).encode()).hexdigest()
+    expected_data = [{
+        "content": content,
+        "meta_data": {
+            "title": mock_result['title'],
+            "uploader": mock_result['uploader'],
+            "upload_date": '20200101',  # Keep as is, matching the `load_data` method
+            "duration": mock_result['duration'],
+            "view_count": mock_result['view_count'],
+            "like_count": mock_result['like_count'],
+            "dislike_count": mock_result['dislike_count'],
+            "average_rating": mock_result['average_rating'],
+            "categories": mock_result['categories'],
+            "tags": mock_result['tags'],
+            "url": video_url,
+            "transcript": serialized_transcript  # Use the serialized transcript here
         }
-    ]
-
-    assert result["data"] == expected_data
+    }]
 
+    assert result['doc_id'] == expected_doc_id
+    assert result['data'] == expected_data
 
 def test_load_data_with_empty_doc(youtube_video_loader):
     video_url = "https://www.youtube.com/watch?v=VIDEO_ID"
-    mock_loader = Mock()
-    mock_loader.load.return_value = []
-
-    with patch("embedchain.loaders.youtube_video.YoutubeLoader.from_youtube_url", return_value=mock_loader):
+    with patch.object(yt_dlp.YoutubeDL, 'extract_info', return_value={'entries': []}):
```

Comment:
Same here can you please use `embedchain.loaders.youtube_video. YoutubeVideoLoader.yt_dlp`